src/genie/libs/parser/iosxe/show_nbar.py

```diff
@@ -0,0 +1,120 @@
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Any, Or, Optional
+import re
+
+# ======================================================
+# Schema for 'show ip nbar protocol-discovery protocol'
+# ======================================================
+
+class ShowIpNbarDiscoverySchema(MetaParser):
+    schema = {
+        'interface': {
+            Any(): {
+                'protocol': {
+                    Any(): {
+                        'IN Packet Count': int,
+                        'OUT Packet Count': int,
+                        'IN Byte Count': int,
+                        'OUT Byte Count': int,
+                        'IN 5min Bit Rate (bps)': int,
+                        'OUT 5min Bit Rate (bps)': int,
+                        'IN 5min Max Bit Rate (bps)': int,
+                        'OUT 5min Max Bit Rate (bps)': int,
+                    }
+                }
+            }
+        }
+    }
+
+class ShowIpNbarDiscovery(ShowIpNbarDiscoverySchema):
+    """Parser for show ip nbar protocol-discovery protocol on IOS-XE
+    parser class - implements detail parsing mechanisms for cli output.
+    """
+    # *************************
+    # schema - class variable
+    #
+    # Purpose is to make sure the parser always return the output
+    # (nested dict) that has the same data structure across all supported
+    # parsing mechanisms (cli(), yang(), xml()).
+    cli_command = 'show ip nbar protocol-discovery protocol'
+
+    
+    def cli(self, output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+            
+        result_dict = {}
+        p1 = re.compile(r'^(?P<interface>Gig.+|Ten.+|Fast.+ |Port.+)')
+        p2 = re.compile(r'^(?P<protocol>[\w\-]+) +(?P<In_Packet_Count>[\d]+) +(?P<Out_Packet_Count>[\d]+)')
+        p3 = re.compile(r'^(?P<In_Byte_Count>[\d]+) +(?P<Out_Byte_Count>[\d]+)')
+        p4 = re.compile(r'^(?P<In_Bitrate>[\d]+) +(?P<Out_Bitrate>[\d]+)')
+        p5 = re.compile(r'^(?P<In_Bitrate_Max>[\d]+) +(?P<Out_Bitrate_Max>[\d]+)')
+
+        
+
+        for line in out.splitlines():
+            #import pdb; pdb.set_trace()
```

Comment:
remove if not needed anymore. line 67 as well.

Reply:
I'll remove the left over pdb lines. Good spot.

Reply:
confirmed the change